tests/test_boltzmann_machine.py

```diff
@@ -78,6 +88,43 @@ def test_forward(self):
             en_boltz = self.bm(fake_spins).item()
             self.assertAlmostEqual(en_bqm, en_boltz, 4)
 
+    def test_estimate_beta(self):
+        s1 = self.sample_1
+        s2 = self.sample_2
+        s3 = torch.vstack([self.sample_2, self.sample_2])
+        bqm = self.bqm
+        self.assertEqual(
+            1.0 / maximum_pseudolikelihood_temperature(bqm, s1.numpy())[0],
+            self.bm.estimate_beta(s1),
+        )
+        self.assertEqual(
+            1.0 / maximum_pseudolikelihood_temperature(bqm, s2.numpy())[0],
+            self.bm.estimate_beta(s2),
+        )
+        self.assertEqual(
+            1.0 / maximum_pseudolikelihood_temperature(bqm, s3.numpy())[0],
+            self.bm.estimate_beta(s3),
+        )
+
+        fake_spins = torch.randn_like(s3)
+        self.assertEqual(
+            1.0 / maximum_pseudolikelihood_temperature(bqm, fake_spins.numpy())[0],
+            self.bm.estimate_beta(fake_spins),
+        )
+
+    def test_compute_expectation_disconnected(self):
+        grbm = GRBM(3, torch.tensor([0, 0, 1]), torch.tensor([1, 2, 2]))
+        grbm.h.data = torch.tensor([-0.1, -0.2, 0.4])
+        grbm.J.data = torch.tensor([-0.7, 0.13, -0.17])
+        beta = 1.337
+        corrupted = torch.tensor([[-1, 1, torch.nan]])
+        expected = grbm.compute_expectation_disconnected(corrupted, beta)[0].tolist()
+        self.assertListEqual(expected[:2], [-1, 1])
+        # -0.13 - 0.17 + 0.3 = 0.1
+        # 0.1 * 1.337 = 0.1337
+        # -tanh(0.1337)
+        self.assertAlmostEqual(expected[-1], -0.132909)
```

Comment:
Pardon my ignorance, but where does `-0.13 - 0.17 + 0.3 = 0.1` come from? Any reason not to do the calculations in code instead of hardcoding it?

Reply:
The expression is derived from the example constructed a few lines above. I'm not sure what the best-practice is so I hard-coded it. I updated the documentation to make the test case more clear.